readium/shared/src/main/java/org/readium/r2/shared/publication/epub/EpubEncryptionParser.kt

```diff
@@ -29,7 +29,7 @@ public object EpubEncryptionParser {
     private fun parseEncryptedData(node: ElementNode): Pair<Url, Encryption>? {
         val resourceURI = node.getFirst("CipherData", Namespaces.ENC)
             ?.getFirst("CipherReference", Namespaces.ENC)?.getAttr("URI")
-            ?.let { Url(it) ?: Url.fromDecodedPath(it) }
```

Comment:
I don't really understand why that was not enough. It should output a canonical path:
- if he input is encoded, then `Url.path` decodes it.
- it the input is decoded, then `Url.fromDecodedPath` encodes it, and then `Url.path` decodes it back.

Reply:
That would work if we used a `String` path in `LcpDecryptor.encryptionData`, but we use a `Url` as it might not be a `path` (remote resources). Then the problem is that you can have the same path encoded differently in the URL, but it still represent the same resource:

* `OEBPS/Epub_LaDeuxi%C3%A8meVie-1.xhtml`
* `OEBPS/Epub_LaDeuxie%CC%80meVie-1.xhtml`

But now that I'm saying this, with LCP we always decrypt packaged (relative) resources, so maybe we could actually use a `String` path in `encryptionData`?

Reply:
I was mistaken with my last comment, the issue is that `é` is valid in an URI, but can still be percent-encoded as `e%CC%80`. So we need the normalization step to have a canonical URI for the comparison.